src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SearchScope.cs

```diff
@@ -112,11 +112,24 @@ public override SymbolReference CreateReference<T>(SearchResult<T> searchResult)
                     _metadataReference);
             }
 
-            protected override Task<IEnumerable<ISymbol>> FindDeclarationsAsync(string name, SymbolFilter filter, SearchQuery searchQuery)
+            protected override async Task<IEnumerable<ISymbol>> FindDeclarationsAsync(string name, SymbolFilter filter, SearchQuery searchQuery)
             {
-                return SymbolFinder.FindDeclarationsAsync(
+                // when searching metadata, give ourselves a limit on how long the search can take.
+                // That way AddImport doesn't pause for a long time while producing metadata indices.
+                const int SearchTimeoutMS = 50;
```

Comment:
We could up it.  On my local machine searching the index took generally sub 1 ms.  But 100ms would certainly be nice and safe.  I like the telemetry idea!